### PR TITLE
[Ecosystem] add a move-crate to simply move dependencies

### PR DESCRIPTION
**Pull request by areshand**
## Motivation
**only migrate e2e test related crates. will have more PR to migrate the rest**

So that we can have a single move dependencies in our crates. 
Future move upgrade will only deal with one crate. 

### Have you read the [Contributing Guidelines on pull requests](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#pull-requests)?

Yes
(You must have submitted a [signed CLA](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#contributor-license-agreement) that includes your GitHub handle prior to us accepting and landing your work. Write your answer here.)

## Test Plan

cargo test -p language-e2e-testsuite

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/aptos-labs/aptos-core/tree/main/developer-docs-site, and link to your PR here.)

### CODE REVIEW

File: aptos-move/move-crates/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+[features]
+default = []
+address32 = ["move-core-types/address32"]
+fuzzing = ["move-core-types/fuzzing", "move-binary-format/fuzzing", "move-vm-types/fuzzing"]
+table-extension = ["move-unit-test/table-extension", "move-vm-test-utils/table-extension"]
+testing = ["move-stdlib/testing", "move-vm-runtime/testing"]
+move-table-extension = ["move-vm-test-utils/move-table-extension"]
+failpoints = ["move-vm-runtime/failpoints"]
+
+
```
Comment (zekun000) [inline aptos-move/move-crates/Cargo.toml:45]: extra blank lines?

**Review by zekun000 — APPROVED** [review]
awesome!

File: aptos-move/move-crates/Cargo.toml (hunk 1 above)
Comment (areshand) [reply aptos-move/move-crates/Cargo.toml]: removed

File: aptos-move/move-crates/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+[package]
+name = "move-crates"
```
Comment (JoshLind) [inline aptos-move/move-crates/Cargo.toml:2]: nit: Maybe have a better description here? e.g., `A custom move crate that wraps the external move dependencies?"

File: aptos-move/move-crates/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+version = "0.0.1"
+authors = ["Aptos Labs <opensource@aptoslabs.com>"]
+repository = "https://github.com/aptos-core/aptos-labs"
+homepage = "https://aptoslabs.com"
+license = "Apache-2.0"
+publish = false
+edition = "2018"
+
+[dependencies]
+move-compiler = { git = "https://github.com/move-language/move", rev = "f2e7585b1ed5bd2810163d6bdebafe5a388881d3" }
```
Comment (JoshLind) [inline aptos-move/move-crates/Cargo.toml:12]: nit: I'm gonna be really annoying and suggest that we keep this list in alphabetical ordering. Especially if we ever have to update it 😄

**Review by JoshLind — COMMENTED** [review]
Awesome! Some nits to hopefully make it clearer 😄

File: aptos-move/move-crates/Cargo.toml (hunk 2 above)
Comment (msmouse) [reply aptos-move/move-crates/Cargo.toml:2]: maybe even a better name.. `move-deps` or even "move-lang"

Comment (areshand) [reply aptos-move/move-crates/Cargo.toml:2]: yes, sir

File: aptos-move/move-crates/Cargo.toml (hunk 3 above)
Comment (areshand) [reply aptos-move/move-crates/Cargo.toml:12]: awesome, sir.

**Review by msmouse — APPROVED** [review]
Thank you ❤️ 

(maybe squash?)

**Pull request merged**